flake.nix

```diff
@@ -62,8 +71,85 @@
               fourmolu
               haskell-language-server
               hlint
-            ]);
+            ])++ [ cardanoPkgs.cardano-node cardanoPkgs.cardano-cli ] ++ [pkgs.postgresql];
         });
+      testnet = perSystem (system:
+        let pkgs =nixpkgsFor system;
+        in
+          pkgs.runCommand "setPrivateNetwork" {buildInputs=(with pkgs; [git]);} (builtins.readFile ./setPrivateNetwork.sh)
+        );
+      runTestnet = self.testnet.x86_64-linux.testnet;
+
+      testnet2 = perSystem (system: 
+        let
+          pkgs = nixpkgsFor system;
+          cardanoPkgs = cardanoPkgsFor system;
+          privateNetwork.url = "https://github.com/woofpool/cardano-private-testnet-setup";
+          localPath = "test-env/ogmios-datum-cache-private-network/cardano-private-testnet-setup";
+        in 
+          pkgs.writeShellScript "runTestnet" ''
+          echo "testnet2 begin"          
+          mkdir -p ${localPath}
+          
+          LOCALREPO_VC_DIR=$LOCALREPO/.git
+          
+          if [ ! -d ${localPath}/.git ]
+          then
+              ${pkgs.git} clone ${privateNetwork.url} ${localPath}
+          fi
+          
+          sleep 1
+          
+          cd ${localPath}
+          ./scripts/automate.sh
+          echo "end"
+          ''
+      );
+
+      run-testnet = perSystem (system:
+        let pkgs =nixpkgsFor system;
+        in
+          pkgs.writeShellScript "run-testnet"  ''make run-testnet''
+        );
+
+      "test-env/ogmios-datum-cache-private-network/cardano-private-testnet-setup"= perSystem (system: 
```

Comment:
If you're trying to clone the repo into a local path called `test-env/ogmios-datum-cache-private-network/cardano-private-testnet-setup`, that's not what this does. Flakes cannot access the local filesystem in general